cmd/kube-proxy/app/server.go

```diff
@@ -682,3 +678,15 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 	}
 	return 0, nil
 }
+
+// CleanupAndExit remove iptables rules and exit if success return nil
+func (s *ProxyServer) CleanupAndExit() error {
+	encounteredError := userspace.CleanupLeftovers(s.IptInterface)
```

Comment:
I wonder if CleanupLeftovers should return the (aggregate) error instead of returning just a boolean.
This would be useful for caller for analysis.

Reply:
I agree that this is a bit ugly but I don't think it'll be easy to fix. A lot of errors from `CleanupLeftovers` don't warrant an early return. For what it's worth, most of the errors are still logged via `klog.Error` though.